Managing modal dialogs in Ember.js with Promises

pichfl opened the pull request.

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
```diff
@@ -0,0 +1,71 @@
+  image: /assets/images/posts/2021-07-13-effective-infrastructure-for-efficient-development-workflows/og-image.png
+---
+
+Modal dialogs are about as widespread as they are missunderstood. No matter if you call them modal windows, popups, popovers, overlays, dialogs: A thing that asks a question or presents a subordinate task; a general annoyance to developers and accessibility experts alike.
```
```suggestion
Modal dialogs are about as widespread as they are missunderstood. No matter if you call them modal windows, popups, popovers, overlays, or dialogs: A thing that asks a question or presents a subordinate task; a general annoyance to developers and accessibility experts alike.
```

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
````diff
@@ -0,0 +1,71 @@
+[Tobias](https://github.com/Turbo87) made Ember Promise Modals so we don't have to implement this on our own. With this addon, can launch any component as a modal, wait for a result, and continue with our apps workflow.
+
+```js
+let result = await this.modals.open('name-of-your-component', { /* passed to the component */ });
````
> `{ /* passed to the component */ }`

Is that the callback? I think it'd be cool to include a real example that shows code that closes the modal.

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
```diff
@@ -0,0 +1,71 @@
+2. Render a modal into the DOM, pass in a callback to resolve the promise
+3. Resolve the promise to unrender the modal and pass back a result
+
+[Tobias](https://github.com/Turbo87) made Ember Promise Modals so we don't have to implement this on our own. With this addon, can launch any component as a modal, wait for a result, and continue with our apps workflow.
```
Should _"Ember Promise Modals"_ not be linked?

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
```diff
@@ -0,0 +1,71 @@
+
+This will trigger a modal dialog, which automatically gains focus for its first focussable element and keyboard accessibility including support for closing the dialog with <kbd>ESC</kbd> as required by [WAI ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) thanks to the included [focus-trap](https://github.com/davidtheclark/focus-trap) integration. It will dim the u ...
+
+The passed component recieve optional data as `@data` and a `@close` action, which will resolve the modal. Anything passed to the action will become the resolved result of the promise, making it easy to communicate data in the preferred Data-Down-Actions-Up pattern of Ember.js. 
```
```suggestion
The passed component recieve optional data as `@data` and a `@close` action, which will close the modal and resolve the promise. Anything passed to the action will become the value the promise resolves with, making it easy to communicate data in the preferred Data-Down-Actions-Up pattern of Ember.js. 
```

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
```diff
@@ -0,0 +1,71 @@
+
+### CSS based animations
+
+For the first official release, [Nick](https://github.com/nickschot) and [myself](https://github.com/pichfl) added native CSS animations using `CSS @keyframes`, which allow for full control over how dialogs and the dimming backdrop appear and disappear on the screen without hogging render performance. A nice and swift default animation is provided out of the box.
```
```suggestion
For the first stable release, [Nick](https://github.com/nickschot) and [myself](https://github.com/pichfl) added native CSS animations using `CSS @keyframes`, which allow for full control over how dialogs and the dimming backdrop appear and disappear on the screen without hogging render performance. A nice and swift default animation is provided out of the box.
```

?

marcoow commented on _posts/2021-08-13-managing-modals-in-ember.md:
```diff
@@ -0,0 +1,71 @@
+
+![Animation of a modal spiraling in after clicking a button below a picture of a cartoon character asking for pictures of Spider Man. The modal shows an image with Spider Man hiding behind a tree and a bold caption saying "I'm Batman"](/assets/images/posts/2021-08-13-managing-modals-in-ember/spiral.gif)
+
+Now it's your turn. Try out [Ember Promise Modals](https://simplabs.github.io/ember-promise-modals/) today!
```
This reads a bit marketing-like tbh :)

pichfl replied:
I'm bad at ending blog posts, next try.

marcoow reviewed (commented):
Nice 👍 

Should the Gif be a Mov instead?

pichfl commented:
@marcoow being faster than ⚡, I added your feedback and replaced the gif with a video, I almost forgot we added video support for Nick's post on web animations.

pichfl commented:
Updated with worked in feedback from Marco and also finalized the Dummy page over at https://github.com/simplabs/ember-promise-modals so this needs just the social media image and a new publishing date and should otherwise be ready for publishing.
